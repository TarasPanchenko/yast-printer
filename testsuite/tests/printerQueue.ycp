/**
 * File:
 *  printerQueue.ycp
 *
 * Module:
 *  Printer configurator
 *
 * Summary:
 *  Printer <=> queue relationship fuctions tests.
 *
 * Authors:
 *  Jiri Srain <jsrain@suse.cz>
 *
 * $Id$
 *
 */

{
    import "Pkg";
    include "testsuite.ycp";

    import "Printer";
    include "printer/misc.ycp";

    Printer::printcap = [$["config":"Epson Stylus Photo 810, Foomatic + gimp-print", "device_ieee":"Stylus Photo 810", "device_db":"STYLUSPHOTO810", "device_id":0, "device_ieee":"Stylus Photo 810", "uri":"parallel:/dev/lp0", "vendor_ieee":"EPSON", "vendor_db":"EPSON", "vendor_id":0, "vendor_ieee":"EPSON", "name":"draft",  "unique_key":"ffnC.s5K9xZzhQb4"],

$[
"config":"Epson Stylus Photo 810, Foomatic + gimp-print", "device_ieee":"Stylus Photo 810", "device_db":"STYLUSPHOTO810", "device_id":0, "device_ieee":"Stylus Photo 810", "name":"draft_gray", "unique_key":"ffnC.s5K9xZzhQb4", "uri":"parallel:/dev/lp0", "vendor_ieee":"EPSON", "vendor_db":"EPSON", "vendor_id":0, "vendor_ieee":"EPSON"],


$["config":"Epson Stylus Photo 810, Foomatic + gimp-print", "device_ieee":"Stylus Photo 810", "device_db":"STYLUSPHOTO810", "device_id":0, "device_ieee":"Stylus Photo 810", "uri":"parallel:/dev/lp0", "vendor_ieee":"EPSON", "vendor_db":"EPSON", "vendor_id":0, "vendor_ieee":"EPSON", "name":"draft",  "unique_key":"1ffnC.s5K9xZzhQb4"],

$[
"config":"Epson Stylus Photo 810, Foomatic + gimp-print", "device_ieee":"Stylus Photo 810", "device_db":"STYLUSPHOTO810", "device_id":0, "device_ieee":"Stylus Photo 810", "name":"draft_gray", "unique_key":"1ffnC.s5K9xZzhQb4", "uri":"parallel:/dev/lp0", "vendor_ieee":"EPSON", "vendor_db":"EPSON", "vendor_id":0, "vendor_ieee":"EPSON"],

];

    TEST (``(getConfiguredPrinters ()), [], nil);
    DUMP ("=============");
    TEST (``(getQueuesOfPrinter ("parallel:/dev/lp0", "EPSON", "Stylus Photo 810")), [], nil);
    DUMP ("===");
    TEST (``(getQueuesOfPrinter ("parallel:/dev/lp0", "EPSON", "STYLUSPHOTO810")), [], nil);
    DUMP ("===");
    TEST (``(getQueuesOfPrinter ("parallel:/dev/lp0", "EPSON", "STYLUSPHOTO820")), [], nil);
    DUMP ("=============");
    TEST (``(isPrinterConfigured ($["vendor_ieee" : "EPSON", "device_ieee" : "Stylus Photo 810", "uri" : "parallel:/dev/lp0"])), [], nil);
    DUMP ("===");
    TEST (``(isPrinterConfigured ($["vendor_ieee" : "EPSON", "device_ieee" : "STYLUSPHOTO810", "uri" : "parallel:/dev/lp0"])), [], nil);
    DUMP ("===");
    TEST (``(isPrinterConfigured ($["vendor_ieee" : "EPSON", "device_ieee" : "Stylus Photo 810", "uri" : "parallel:/dev/lp1"])), [], nil);
    DUMP ("===");
    TEST (``(isPrinterConfigured ($[])), [], nil);
    DUMP ("===");
    TEST (``(isPrinterConfigured ($["unique_key":"ffnC.s5K9xZzhQb4"])), [], nil);
    DUMP ("==========================================");

    Printer::printcap = add (Printer::printcap,

$[

"config":"Epson Stylus Photo 820, Foomatic + gimp-print", "device_ieee":"Stylus Photo 820", "device_db":"STYLUSPHOTO820", "device_id":0, "device_ieee":"Stylus Photo 820", "uri":"parallel:/dev/lp0", "vendor_ieee":"EPSON", "vendor_db":"EPSON", "vendor_id":0, "vendor_ieee":"EPSON", "name":"draft",

]);

    TEST (``(getConfiguredPrinters ()), [], nil);
    DUMP ("=============");
    TEST (``(getQueuesOfPrinter ("parallel:/dev/lp0", "EPSON", "Stylus Photo 810")), [], nil);
    DUMP ("===");
    TEST (``(getQueuesOfPrinter ("parallel:/dev/lp0", "EPSON", "STYLUSPHOTO810")), [], nil);
    DUMP ("===");
    TEST (``(getQueuesOfPrinter ("parallel:/dev/lp0", "EPSON", "STYLUSPHOTO820")), [], nil);
    DUMP ("=============");
    TEST (``(isPrinterConfigured ($["vendor_ieee" : "EPSON", "device_ieee" : "Stylus Photo 810", "uri" : "parallel:/dev/lp0"])), [], nil);
    DUMP ("===");
    TEST (``(isPrinterConfigured ($["vendor_ieee" : "EPSON", "device_ieee" : "STYLUSPHOTO810", "uri" : "parallel:/dev/lp0"])), [], nil);
    DUMP ("===");
    TEST (``(isPrinterConfigured ($["vendor_ieee" : "EPSON", "device_ieee" : "Stylus Photo 810", "uri" : "parallel:/dev/lp1"])), [], nil);
    DUMP ("===");
    TEST (``(isPrinterConfigured ($[])), [], nil);
    DUMP ("===");
    TEST (``(isPrinterConfigured ($["unique_key":"nonsense"])), [], nil);
    DUMP ("==========================================");


}
