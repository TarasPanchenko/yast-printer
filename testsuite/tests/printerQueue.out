Read	.target.size "/etc/install.inf" -1
Read	.target.tmpdir "/tmp/"
Read	.target.tmpdir "/tmp/"
Execute	.target.bash "/usr/lib/YaST2/bin/version" 0
Read	.sysconfig.language.RC_LANG ""
Read	.sysconfig.language.DEFAULT_LANGUAGE ""
Read	.target.yast2 "lang2iso.ycp" nil
Read	.target.size "/etc/sysconfig/language" -1
Read	.sysconfig.console.CONSOLE_FONT ""
Read	.sysconfig.console.CONSOLE_SCREENMAP ""
Read	.sysconfig.console.CONSOLE_UNICODEMAP ""
Read	.sysconfig.console.CONSOLE_MAGIC ""
Read	.sysconfig.console.CONSOLE_ENCODING ""
Read	.sysconfig.language.RC_LANG ""
Read	.probe.architecture "i386"
Read	.probe.has_apm true
Read	.probe.has_pcmcia false
Read	.probe.has_smp false
Read	.probe.system []
Return	[$["config":"Epson Stylus Photo 810, Foomatic + gimp-print", "device_db":"STYLUSPHOTO810", "device_id":0, "device_ieee":"Stylus Photo 810", "name":"draft", "unique_key":"ffnC.s5K9xZzhQb4", "uri":"parallel:/dev/lp0", "vendor_db":"EPSON", "vendor_id":0, "vendor_ieee":"EPSON"]]
Dump	=============
Return	[$["config":"Epson Stylus Photo 810, Foomatic + gimp-print", "device_db":"STYLUSPHOTO810", "device_id":0, "device_ieee":"Stylus Photo 810", "name":"draft", "unique_key":"ffnC.s5K9xZzhQb4", "uri":"parallel:/dev/lp0", "vendor_db":"EPSON", "vendor_id":0, "vendor_ieee":"EPSON"], $["config":"Epson Stylus Photo 810, Foomatic + gimp-print", "device_db":"STYLUSPHOTO810", "device_id":0, "device_ieee":"Stylus Photo 810", "name":"draft_gray", "unique_key":"ffnC.s5K9xZzhQb4", "uri":"parallel:/dev/lp0", "vendor_db":"EPSON", "vendor_id":0, "vendor_ieee":"EPSON"], $["config":"Epson Stylus Photo 810, Foomatic + gimp-print", "device_db":"STYLUSPHOTO810", "device_id":0, "device_ieee":"Stylus Photo 810", "name":"draft", "unique_key":"1ffnC.s5K9xZzhQb4", "uri":"parallel:/dev/lp0", "vendor_db":"EPSON", "vendor_id":0, "vendor_ieee":"EPSON"], $["config":"Epson Stylus Photo 810, Foomatic + gimp-print", "device_db":"STYLUSPHOTO810", "device_id":0, "device_ieee":"Stylus Photo 810", "name":"draft_gray", "unique_key":"1ffnC.s5K9xZzhQb4", "uri":"parallel:/dev/lp0", "vendor_db":"EPSON", "vendor_id":0, "vendor_ieee":"EPSON"]]
Dump	===
Return	[$["config":"Epson Stylus Photo 810, Foomatic + gimp-print", "device_db":"STYLUSPHOTO810", "device_id":0, "device_ieee":"Stylus Photo 810", "name":"draft", "unique_key":"ffnC.s5K9xZzhQb4", "uri":"parallel:/dev/lp0", "vendor_db":"EPSON", "vendor_id":0, "vendor_ieee":"EPSON"], $["config":"Epson Stylus Photo 810, Foomatic + gimp-print", "device_db":"STYLUSPHOTO810", "device_id":0, "device_ieee":"Stylus Photo 810", "name":"draft_gray", "unique_key":"ffnC.s5K9xZzhQb4", "uri":"parallel:/dev/lp0", "vendor_db":"EPSON", "vendor_id":0, "vendor_ieee":"EPSON"], $["config":"Epson Stylus Photo 810, Foomatic + gimp-print", "device_db":"STYLUSPHOTO810", "device_id":0, "device_ieee":"Stylus Photo 810", "name":"draft", "unique_key":"1ffnC.s5K9xZzhQb4", "uri":"parallel:/dev/lp0", "vendor_db":"EPSON", "vendor_id":0, "vendor_ieee":"EPSON"], $["config":"Epson Stylus Photo 810, Foomatic + gimp-print", "device_db":"STYLUSPHOTO810", "device_id":0, "device_ieee":"Stylus Photo 810", "name":"draft_gray", "unique_key":"1ffnC.s5K9xZzhQb4", "uri":"parallel:/dev/lp0", "vendor_db":"EPSON", "vendor_id":0, "vendor_ieee":"EPSON"]]
Dump	===
Return	[]
Dump	=============
Return	true
Dump	===
Return	false
Dump	===
Return	false
Dump	===
Return	false
Dump	===
Return	true
Dump	==========================================
Return	[$["config":"Epson Stylus Photo 810, Foomatic + gimp-print", "device_db":"STYLUSPHOTO810", "device_id":0, "device_ieee":"Stylus Photo 810", "name":"draft", "unique_key":"ffnC.s5K9xZzhQb4", "uri":"parallel:/dev/lp0", "vendor_db":"EPSON", "vendor_id":0, "vendor_ieee":"EPSON"], $["config":"Epson Stylus Photo 820, Foomatic + gimp-print", "device_db":"STYLUSPHOTO820", "device_id":0, "device_ieee":"Stylus Photo 820", "name":"draft", "uri":"parallel:/dev/lp0", "vendor_db":"EPSON", "vendor_id":0, "vendor_ieee":"EPSON"]]
Dump	=============
Return	[$["config":"Epson Stylus Photo 810, Foomatic + gimp-print", "device_db":"STYLUSPHOTO810", "device_id":0, "device_ieee":"Stylus Photo 810", "name":"draft", "unique_key":"ffnC.s5K9xZzhQb4", "uri":"parallel:/dev/lp0", "vendor_db":"EPSON", "vendor_id":0, "vendor_ieee":"EPSON"], $["config":"Epson Stylus Photo 810, Foomatic + gimp-print", "device_db":"STYLUSPHOTO810", "device_id":0, "device_ieee":"Stylus Photo 810", "name":"draft_gray", "unique_key":"ffnC.s5K9xZzhQb4", "uri":"parallel:/dev/lp0", "vendor_db":"EPSON", "vendor_id":0, "vendor_ieee":"EPSON"], $["config":"Epson Stylus Photo 810, Foomatic + gimp-print", "device_db":"STYLUSPHOTO810", "device_id":0, "device_ieee":"Stylus Photo 810", "name":"draft", "unique_key":"1ffnC.s5K9xZzhQb4", "uri":"parallel:/dev/lp0", "vendor_db":"EPSON", "vendor_id":0, "vendor_ieee":"EPSON"], $["config":"Epson Stylus Photo 810, Foomatic + gimp-print", "device_db":"STYLUSPHOTO810", "device_id":0, "device_ieee":"Stylus Photo 810", "name":"draft_gray", "unique_key":"1ffnC.s5K9xZzhQb4", "uri":"parallel:/dev/lp0", "vendor_db":"EPSON", "vendor_id":0, "vendor_ieee":"EPSON"]]
Dump	===
Return	[$["config":"Epson Stylus Photo 810, Foomatic + gimp-print", "device_db":"STYLUSPHOTO810", "device_id":0, "device_ieee":"Stylus Photo 810", "name":"draft", "unique_key":"ffnC.s5K9xZzhQb4", "uri":"parallel:/dev/lp0", "vendor_db":"EPSON", "vendor_id":0, "vendor_ieee":"EPSON"], $["config":"Epson Stylus Photo 810, Foomatic + gimp-print", "device_db":"STYLUSPHOTO810", "device_id":0, "device_ieee":"Stylus Photo 810", "name":"draft_gray", "unique_key":"ffnC.s5K9xZzhQb4", "uri":"parallel:/dev/lp0", "vendor_db":"EPSON", "vendor_id":0, "vendor_ieee":"EPSON"], $["config":"Epson Stylus Photo 810, Foomatic + gimp-print", "device_db":"STYLUSPHOTO810", "device_id":0, "device_ieee":"Stylus Photo 810", "name":"draft", "unique_key":"1ffnC.s5K9xZzhQb4", "uri":"parallel:/dev/lp0", "vendor_db":"EPSON", "vendor_id":0, "vendor_ieee":"EPSON"], $["config":"Epson Stylus Photo 810, Foomatic + gimp-print", "device_db":"STYLUSPHOTO810", "device_id":0, "device_ieee":"Stylus Photo 810", "name":"draft_gray", "unique_key":"1ffnC.s5K9xZzhQb4", "uri":"parallel:/dev/lp0", "vendor_db":"EPSON", "vendor_id":0, "vendor_ieee":"EPSON"]]
Dump	===
Return	[$["config":"Epson Stylus Photo 820, Foomatic + gimp-print", "device_db":"STYLUSPHOTO820", "device_id":0, "device_ieee":"Stylus Photo 820", "name":"draft", "uri":"parallel:/dev/lp0", "vendor_db":"EPSON", "vendor_id":0, "vendor_ieee":"EPSON"]]
Dump	=============
Return	true
Dump	===
Return	false
Dump	===
Return	false
Dump	===
Return	false
Dump	===
Return	false
Dump	==========================================
